Sources/CodeEditSourceEditor/Controller/TextViewController.swift

```diff
@@ -18,6 +18,76 @@ import TextFormation
 /// tree-sitter for syntax highlighting, and TextFormation for live editing completions.
 /// 
 public class TextViewController: NSViewController {
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        NSEvent.addLocalMonitorForEvents(matching: .keyDown) {
+            self.keyDown(with: $0)
+            return $0
+        }
+    }
+
+    /// key Down event recognizer, currently only set up to recognize CMD + /
+    override public func keyDown(with event: NSEvent) {
+        let charactersIgnoringModifiers = event.charactersIgnoringModifiers
+        let commandKey = NSEvent.ModifierFlags.command.rawValue
+        let modifierFlags = event.modifierFlags.intersection(.deviceIndependentFlagsMask).rawValue
+        if modifierFlags == commandKey && event.charactersIgnoringModifiers == "/" {
+            commandSlashCalled()
+        } else {
+            super.keyDown(with: event)
+        }
+    }
+
+    /// Method called when CMD + / key sequence recognized, comments cursor's current line of code
+    func commandSlashCalled() {
+        print(cursorPositions)
+        guard let cursorPosition = cursorPositions.first else {
+            print("There is no cursor \(#function)")
+            return
+        }
+        guard let lineInfo = textView.layoutManager.textLineForIndex(cursorPosition.line - 1) else {
+            print("There are no characters/lineInfo \(#function)")
+            return
+        }
+        let lineFirstCharIndex = lineInfo.range.location
+        // lineComment only required at front of line
+        if !language.lineCommentString.isEmpty {
```

Comment:
Can we get more comments on what this condition is for and why we have it? For example, because some languages like HTML require a start and ending comment string and doesnt have a `lineCommentString` value